src/msg.rs

```diff
@@ -6,27 +6,45 @@ use serde::{Deserialize, Serialize};
 pub struct InstantiateMsg {
     pub vault_denom: String,
     pub amount_to_stake_per_validator: Uint128,
-    pub max_records_to_update_per_run: u32,
+    pub batch_size: u32,
 }
 
 #[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
 #[serde(rename_all = "snake_case")]
 pub enum ExecuteMsg {
     RecordMetrics { timestamp: u64 }, // 12:00AM
     AddValidator { addr: Addr },
-    UpdateRecordsToUpdatePerRun { no: u32 },
+    UpdateConfig { batch_size: u32 },
 }
 
 #[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
 #[serde(rename_all = "snake_case")]
 pub enum QueryMsg {
-    GetHistoryByTime {
+    GetAllTimestamps {},
```

Comment:
Good job with these projections. 

Next time, think of consolidating them by taking in Vec<Addr> or Vec<Timestamp>. Don't need this change now.